src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractMemberScopedReferenceFinder.cs

```diff
@@ -99,13 +100,22 @@ protected sealed override bool CanFind(TSymbol symbol)
             CancellationToken cancellationToken)
         {
             var service = state.Document.GetRequiredLanguageService<ISymbolDeclarationService>();
-            var tokens = service.GetDeclarations(container)
-                .SelectMany(r => r.GetSyntax(cancellationToken)
-                    .DescendantTokens()
-                    .Where(t => TokensMatch(state, t, symbol.Name)))
-                .ToImmutableArray();
+            using var _ = ArrayBuilder<SyntaxToken>.GetInstance(out var tokens);
 
-            return FindReferencesInTokensAsync(symbol, state, tokens, cancellationToken);
+            foreach (var declaration in service.GetDeclarations(container))
+            {
+                var syntax = declaration.GetSyntax(cancellationToken);
+                if (syntax.SyntaxTree != state.SyntaxTree)
+                    continue;
```

Comment:
this is hte added code.  A container may show up in multiple files (esp. in error cases).  We want the one that matches the file we're currently looking within.